cms/workflows/static/js/workflow-tweaks.js

```diff
@@ -0,0 +1,43 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const editForm = document.getElementById("page-edit-form");
+
+    if (!editForm) {
+        return; // Exit if the edit form is not found
+    }
+
+    // Store the initial form payload on load
+    const initialFormPayload = new URLSearchParams(new FormData(editForm)).toString();
+
+    // Function to toggle approve buttons based on form changes
+    function toggleApproveButtons() {
+        const newPayload = new URLSearchParams(new FormData(editForm)).toString();
+        const approveButtons = document.querySelectorAll('[data-workflow-action-name="approve"]');
+
+        if (initialFormPayload !== newPayload) {
+            // Disable the approve buttons when form has changed
+            approveButtons.forEach(button => {
+                button.setAttribute('disabled', 'disabled');
+                button.classList.add('disabled');
+            });
+        } else {
+            // Re-enable the buttons when form returns to initial state
+            approveButtons.forEach(button => {
+                button.removeAttribute('disabled');
+                button.classList.remove('disabled');
+            });
+        }
+    }
+
+    // Add event listeners to form elements to detect changes
+    const formElements = editForm.querySelectorAll('input, textarea, select');
+    formElements.forEach(element => {
+        element.addEventListener('change', toggleApproveButtons);
+        element.addEventListener('input', toggleApproveButtons);
+    });
+
+    // Also listen for changes that might be triggered by JavaScript
+    editForm.addEventListener('change', toggleApproveButtons);
```

Comment:
Events will [bubble up](https://developer.mozilla.org/en-US/docs/Learn_web_development/Core/Scripting/Event_bubbling) so we can just listen on the `editForm` (it's also more efficient as we only add two event listeners as opposed to 2n + 1):
```suggestion
    editForm.addEventListener('change', toggleApproveButtons);
    editForm.addEventListener('input', toggleApproveButtons);
```